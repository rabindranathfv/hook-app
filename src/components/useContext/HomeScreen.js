import React, { useContext } from 'react'
import { UserContext } from './UserContest'

export const HomeScreen = () => {

    const { user, setUser } = useContext(UserContext)

    return (
        <div>
           <h1> Home Screen </h1>
            <hr /> 

            <pre> 
                { JSON.stringify(user, null, 3) }
            </pre>
        </div>
    )
}
