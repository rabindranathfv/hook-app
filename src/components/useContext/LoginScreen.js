import React, { useContext, useEffect } from 'react';
import { UserContext } from './UserContest';

export const LoginScreen = () => {
    
    const { user, setUser } = useContext(UserContext);

    useEffect(() => {
        console.log('Listem all user changes')
    }, [user])
    

    const handleInputChange = (e) => {
        console.log(user);
        setUser({
            ...user,
            [ e.target.name ] : e.target.value
    
        })
    };

    return (
        <div>
            <h1> Login Screen </h1>
            <hr />

            <div className="form-group">
                <label htmlFor=""></label>
                <input type="text"
                className="form-control" name="name" id="" aria-describedby="name-form" placeholder="enter your name" value={user?.name} onChange={handleInputChange}/>
                <small className="username info">Name</small>

                <label htmlFor=""></label>
                <input type="text"
                className="form-control" name="email" id="" aria-describedby="email-form" placeholder="enter your email" value={user?.email} onChange={handleInputChange}/>
                <small className="email info">Email</small>
            </div>
        </div>
    )
}
