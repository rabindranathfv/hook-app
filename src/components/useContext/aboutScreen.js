import React, { useContext } from 'react'
import { UserContext } from './UserContest';

export const AboutScreen = () => {

    const { user, setUser } = useContext(UserContext);

    return (
        <div>
            <h1> About Screen </h1>
            <hr />

            <h3>
                welcome { user?.name }!
            </h3>
        </div>
    )
}
