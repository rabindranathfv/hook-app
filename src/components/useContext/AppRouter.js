import React from 'react';
import {
    BrowserRouter as Router,
    Route,
    Routes,
    useRouteMatch,
    useParams
} from 'react-router-dom';

import { NavBar } from './NavBar';

import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';

export const AppRouter = () => {
    return (
        <Router>
            <div>

                <NavBar />

                <Routes>
                    <Route path="/about" element={<AboutScreen />} />
                    <Route path="/login" element={<LoginScreen />} />
                    <Route path="/" element={<HomeScreen />} />
                    <Route path="*" element={<HomeScreen />} />
                </Routes>
            </div>
        </Router>
    )
}
