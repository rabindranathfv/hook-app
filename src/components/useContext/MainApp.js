import React, { useState } from 'react'
import { AppRouter } from './AppRouter';
import { UserContext } from './UserContest';


export const MainApp = () => {
    
    const [user, setUser] = useState({ name: '', email: ''})
    
    return (
        <UserContext.Provider value={ { user, setUser } }>
            <AppRouter />
        </UserContext.Provider>
    )
}
