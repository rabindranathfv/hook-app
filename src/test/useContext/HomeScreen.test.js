import React from 'react';
import { shallow, mount } from 'enzyme';
import { HomeScreen } from '../../components/useContext/HomeScreen';
import { UserContext } from '../../components/useContext/UserContest';

const user = {
    name: 'rabindranath',
    email: 'test@gmail.com'
}

describe('HomeScreen', () => {

    const wrapper = mount( 
    <UserContext.Provider value={{ user }}>
        <HomeScreen  />    
    </UserContext.Provider>)
    it('Should render HomeScreen', () => {
        expect(wrapper).toMatchSnapshot();
    });
});