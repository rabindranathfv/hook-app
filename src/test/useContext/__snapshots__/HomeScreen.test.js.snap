// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeScreen Should render HomeScreen 1`] = `
<div>
  <h1>
     Home Screen 
  </h1>
  <hr />
  <pre>
    {
   "name": "rabindranath",
   "email": "test@gmail.com"
}
  </pre>
</div>
`;
